lib/src/error_tracking/dart_exception_processor.dart

```diff
@@ -0,0 +1,335 @@
+  static Map<String, dynamic> processException({
+    required Object error,
+    StackTrace? stackTrace,
+    Map<String, Object>? properties,
+    bool handled = true,
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+    StackTrace Function()? stackTraceProvider, //for testing
+  }) {
+    StackTrace? effectiveStackTrace = stackTrace;
+    bool isGeneratedStackTrace = false;
+
+    // If it's an Error, try to use its built-in stackTrace
+    if (error is Error) {
+      effectiveStackTrace ??= error.stackTrace;
+    }
+
+    // If still null or empty, get current stack trace
+    if (effectiveStackTrace == null ||
+        effectiveStackTrace == StackTrace.empty) {
+      effectiveStackTrace = stackTraceProvider?.call() ?? StackTrace.current;
+      isGeneratedStackTrace = true; // Flag to remove top PostHog frames
+    }
+
+    // Process single exception (Dart doesn't provide standard exception chaining afaik)
+    final frames = _parseStackTrace(
+      effectiveStackTrace,
+      inAppIncludes: inAppIncludes,
+      inAppExcludes: inAppExcludes,
+      inAppByDefault: inAppByDefault,
+      removeTopPostHogFrames: isGeneratedStackTrace,
+    );
+
+    // we consider primitives and generated Strack traces as synthetic
+    final exceptionData = <String, dynamic>{
+      'type': _getExceptionType(error),
+      'mechanism': {
+        'handled': handled,
+        'synthetic': _isPrimitive(error) || isGeneratedStackTrace,
+        'type': 'generic',
+      },
+      'thread_id': _getCurrentThreadId(),
+    };
+
+    // Add exception message, if available
+    final errorMessage = error.toString();
+    if (errorMessage.isNotEmpty) {
+      exceptionData['value'] = errorMessage;
+    }
+
+    // Add module from first stack frame (where exception was thrown)
+    final exceptionModule = _getExceptionModule(effectiveStackTrace);
+    if (exceptionModule != null && exceptionModule.isNotEmpty) {
+      exceptionData['module'] = exceptionModule;
+    }
+
+    // Add package from first stack frame (where exception was thrown)
+    final exceptionPackage = _getExceptionPackage(effectiveStackTrace);
+    if (exceptionPackage != null && exceptionPackage.isNotEmpty) {
+      exceptionData['package'] = exceptionPackage;
+    }
+
+    // Add stacktrace, if any frames are available
+    if (frames.isNotEmpty) {
+      exceptionData['stacktrace'] = {
+        'frames': frames,
+        'type': 'raw',
+      };
+    }
+
+    final result = <String, dynamic>{
+      '\$exception_level': handled ? 'error' : 'fatal',
+      '\$exception_list': [exceptionData],
+    };
+
+    // Add custom properties if provided
+    if (properties != null) {
+      for (final entry in properties.entries) {
+        // Don't allow overwriting system properties
+        if (!result.containsKey(entry.key)) {
+          result[entry.key] = entry.value;
+        }
```

Comment:
I'd say the given props always have precedence, like we do for all analytics events etc, so just merge the given props last, and if users set something internal/system props, that was their intention, similar to the beforeSend callback, they can change anything